### PR TITLE
Don't delete the PU if its present in the cluster.

### PR DESCRIPTION
**Pull request by abhijitherekar**
Also change the log level to warn.

#### Description
*Changes proposed in this pull request.*

#### Test plan
*Outline the test plan used to test this change before merging it.*

> Fixes #.

### CODE REVIEW

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "954",
    "commit-sha": "8131efa5d5eb862d5d4ef1678cb3757edcfb05d6",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "mtls",
    "component": "enforcerd",
    "pr-id": "1502",
    "commit-sha": "3c7dc5a6fed8856535c925b3070e0aac780a6689",
    "pipeline": "master"
  },
  {
    "project": "mtls",
    "component": "trireme-lib",
    "pr-id": "954",
    "commit-sha": "8131efa5d5eb862d5d4ef1678cb3757edcfb05d6",
    "pipeline": "master"
  }
]
```

File: monitor/internal/pod/delete_controller.go [hunk 1]
```diff
@@ -142,16 +143,19 @@ func deleteControllerProcessItem(backgroundCtx context.Context, c client.Client,
...
-			ctx,
-			podUID,
-			common.EventDestroy,
-			policy.NewPURuntimeWithDefaults(),
-		); err != nil {
-			// we don't really care, we just warn
-			zap.L().Warn("DeleteController: Failed to handle destroy event", zap.String("puID", podUID), zap.String("namespacedName", req.String()), zap.Error(err))
-		}
+		// NODE REBOOT can make us land here.
+		// 1. Don't send a delete event as we know that the pod is present in cluster.
```
Comment (mheese) [inline monitor/internal/pod/delete_controller.go]: We must delete the PU. PUs are bound to the enforcers that manage them. So even if the same pod moved to a new node, this must entail a PUCreate and PUStart on the other host, and a PUDestroy on this node.

File: monitor/internal/pod/delete_controller.go [hunk 2]
```diff
@@ -142,16 +143,19 @@ func deleteControllerProcessItem(backgroundCtx context.Context, c client.Client,
...
-		if err := pc.Policy.HandlePUEvent(
-			ctx,
-			podUID,
-			common.EventDestroy,
-			policy.NewPURuntimeWithDefaults(),
-		); err != nil {
-			// we don't really care, we just warn
-			zap.L().Warn("DeleteController: Failed to handle destroy event", zap.String("puID", podUID), zap.String("namespacedName", req.String()), zap.Error(err))
-		}
+		// NODE REBOOT can make us land here.
```
Comment (mheese) [inline monitor/internal/pod/delete_controller.go]: This assumption sounds interesting ... why do you think so, and why would it cause problems?

Comment (abhijitherekar) [reply monitor/internal/pod/delete_controller.go]: I will get rid of this, my assumption was wrong.

**Review by mheese — APPROVED** [review]
cleaner messages and better log levels. Thanks

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "mtls",
    "component": "enforcerd",
    "pr-id": "1502",
    "commit-sha": "3c7dc5a6fed8856535c925b3070e0aac780a6689",
    "pipeline": "master"
  },
  {
    "project": "mtls",
    "component": "trireme-lib",
    "pr-id": "954",
    "commit-sha": "d37cc5d5362bc48ecfed035e91f79bc00e0828e0",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "mtls",
    "component": "trireme-lib",
    "pr-id": "954",
    "commit-sha": "d37cc5d5362bc48ecfed035e91f79bc00e0828e0",
    "pipeline": "master"
  },
  {
    "project": "mtls",
    "component": "enforcerd",
    "pr-id": "1502",
    "commit-sha": "ff425c592dd4058ecf2ac80993f6be454d541282",
    "pipeline": "master"
  }
]
```

**Pull request merged**